src/storage/storage.ts

```diff
@@ -1,46 +1,44 @@
-import { ActivityPub } from "../activityPub/activityPub";
-import { BatchFileObject } from "../batch/batch";
-// import * as config from "../config/config";
-import { Config } from "../config/config";
-import { NotImplementedError } from "../utilities/errors";
+import { Config, getConfig } from "../config/config";
+import { MissingStore } from "../utilities/errors";
 
-export type File = BatchFileObject | ActivityPub;
+export type File = Buffer | string;
+export type Content = string | Buffer;
 
 export interface StorageInterface {
-  store(file: File): Promise<string>;
-  fetch(uri: string): Promise<File>;
+  put(targetPath: string, content: Content): Promise<URL>;
+  get(targetPath: string): Promise<File>;
```

Comment:
This is not really related to this I wonder how we are going to use get? I like that fetch is gone. URI fetching should not happen here. I guess this could be used for recovering only the files you stored via put.

Reply:
It can also be used by archivers for storage, and indexers who want to get new batches and read/parse from local storage.

Reply:
Yeah, I agree with this. I asked this question to Shannon yesterday. But left it just in case I missed something.

Reply:
looks like Shannon answered question.